aiohomekit/controller/controller.py

```diff
@@ -113,12 +114,28 @@ async def async_start(self) -> None:
     async def async_stop(self) -> None:
         await self._tasks.aclose()
 
-    async def async_find(self, device_id: str) -> AbstractDiscovery:
+    async def async_find(
+        self, device_id: str, timeout: float = 30.0
+    ) -> AbstractDiscovery:
+        pending = []
         for transport in self._transports:
-            try:
-                return await transport.async_find(device_id)
-            except AccessoryNotFoundError:
-                pass
+            pending.append(asyncio.create_task(transport.async_find(device_id, timeout)))
+
+        try:
+            while pending:
+                done, pending = await asyncio.wait(pending, return_when=asyncio.FIRST_COMPLETED)
+                for result in done:
+                    try:
+                        return result.result()
+                    except AccessoryNotFoundError:
+                        continue
+        finally:
+            [task.cancel() for task in pending]
+            for task in pending:
+                try:
```

Comment:
Could use `with contextlib.suppress(asyncio.CancelledError)` 🤷‍♂️ but doesn't really matter

Reply:
I was thinking of making a async_cancel(list[Task]) that did this because it's something I write a lot.